### PR TITLE
Removed DoubleAmpersand and DoublePipe IRs

### PR DESCRIPTION
**Pull request by jigold**
Wasn't short circuiting properly as written. Caused this test to fail:

`self.assertEqual(hl.cond(hl.null(hl.tbool), 1, 2, missing_false=True).value, 2)`

### CODE REVIEW

File: src/main/scala/is/hail/expr/ir/BinaryOp.scala [hunk 1]
```diff
@@ -108,8 +107,6 @@ object BinaryOp {
       val ll = coerce[Boolean](l)
       val rr = coerce[Boolean](r)
       op match {
-        case DoubleAmpersand() => ll && rr
-        case DoublePipe() => ll || rr
```
Comment (danking) [inline src/main/scala/is/hail/expr/ir/BinaryOp.scala]: A short-circuiting implementation would be:
```
ll.mux(true, rr)
```

Comment (catoverdrive) [reply src/main/scala/is/hail/expr/ir/BinaryOp.scala]: no, because the missingness is checked before it reaches this point, right?

Comment (catoverdrive) [reply src/main/scala/is/hail/expr/ir/BinaryOp.scala]: we'd want to either fix `ApplyBinaryPrimOp` to not always return `ml || mr` as the missingness, or to define a separate thing for cases like this.

Comment (catoverdrive) [reply src/main/scala/is/hail/expr/ir/BinaryOp.scala]: `(ll || rr)` is already a short-circuit operator in the traditional sense, but it doesn't handle missingness here.

**Comment by danking** [comment]
@jigold ok let's ditch this for now and come back to it. Sounds like @catoverdrive has some ideas for a better approach to binops

**Pull request merged**